homeassistant/components/wolflink/__init__.py

```diff
@@ -134,7 +134,7 @@ async def fetch_parameters_init(client: WolfClient, gateway_id: int, device_id:
     """Fetch all available parameters with usage of WolfClient but handles all exceptions and results in ConfigEntryNotReady."""
     try:
         return await fetch_parameters(client, gateway_id, device_id)
-    except (ConnectError, ConnectTimeout, FetchFailed) as exception:
+    except (ConnectError, ConnectTimeout, FetchFailed, ReadTimeout) as exception:
```

Comment:
Maybe just except `httpx.TransportError` instead? That includes all the currently caught httpx errors.

Reply:
Considering, that `fetch_parameters_init` method only raises `ConfigEntryNotReady` on exceptions, it might also be `httpx.RequestError` or even `httpx.HTTPError` in general. In any of the cases, it will be not ready and will log the exception. Isn't it?

Reply:
Yeah, I haven't looked more into what the library does but the other httpx errors can probably happen too unless the library handles them.

Reply:
I think that using `httpx.RequestError` makes sense more sense here. Using `httpx.HTTPError` (containing `httpx.HTTPStatusError`) might be problematic, there are http status codes that should not be retried, for example 403 (forbidden).

The same ihmo holds for `async_update_data`, it makes more sense to use `httpx.RequestError` than `httpx.ConnectError`, so I have updated it there as well.